src/entt/entity/handle.hpp

```diff
@@ -47,19 +69,64 @@ struct basic_handle {
     }
 
     /**
-     * @brief Assigns the null object to a handle.
+    * @brief Assigns a registry and an entity to a handle.
+    * @param other Another handle.
+    * @return This handle.
+    */
+    basic_handle & operator=(const basic_handle &other) ENTT_NOEXCEPT {
+        reg = other.reg;
+        entt = other.entt;
+        return *this;
+    }
+
+    /**
+     * @brief Assigns the null object to a handle, invalidating both registry and entity.
      * @return This handle.
      */
     basic_handle & operator=(null_t) ENTT_NOEXCEPT {
-        return (*this = static_cast<entity_type>(null));
+        reg = nullptr;
+        entt = null;
```

Comment:
Probably this one was wrong already. We should cast the null entity to the underlying type of the entity type, then construct an instance. If it seems tricky, it helps to fully support custom identifiers, also the trickiest ones (and ofc I've realized this when I saw the trickiest ones at work! 😄).

Reply:
Will do!